pkgs/development/interpreters/octave/default.nix

```diff
@@ -1,4 +1,6 @@
-{ stdenv
+{ pkgs
```

Comment:
In general, using `pkgs` like that inside expressions increases eval time. See: https://github.com/NixOS/nixpkgs/issues/38635#issuecomment-379678500

But if you wish to design `octave-packages.nix` in a way that will not put into scope all packages from `top-level.nix` (differently than what I did in https://github.com/NixOS/nixpkgs/pull/106599 ), than that's justified.

Reply:
Interesting to learn. However, `pkgs` is needed for `octavePackages` to receive their potential inputs that are not already present in Octave. For example, the `io` package requires the `nettle` library for some applications. `parallel` needs `gnutls`, which Octave does not currently have. 

Is there a better way to make sure packages get the dependencies they need (that are not also used by Octave itself)?